src/mlx/warnings.py

```diff
@@ -13,45 +14,99 @@
 
 
 class WarningsPlugin:
-    def __init__(self):
-        self.sphinx_counter = 0
-        self.doxygen_counter = 0
-        self.junit_counter = 0
 
-    def check_sphinx_warnings(self, line):
+    def __init__(self, sphinx = False, doxygen = False, junit = False):
+        # type: (boolean, boolean, boolean) -> None
         '''
-        Function for counting the number of sphinx warnings in a logfile.
-        The function returns the number of warnings found
-        '''
-        if re.search(sphinx_pattern, line):
-            self.sphinx_counter += 1
+        Function for initializing the parsers
 
-    def return_sphinx_warnings(self):
-        print("{count} sphinx warnings found".format(count=self.sphinx_counter))
-        return self.sphinx_counter
-
-    def check_doxygen_warnings(self, line):
+        Args:
+            sphinx          enable sphinx parser
```

Comment:
Could you write as ```sphinx (bool):   enable sphinx parser```? http://www.sphinx-doc.org/en/stable/ext/example_google.html#example-google